runner-starter/src/main/java/org/apache/apisix/plugin/runner/PluginRunnerApplication.java

```diff
@@ -17,17 +17,138 @@
+import org.springframework.context.ApplicationContext;
+import org.springframework.scheduling.annotation.EnableScheduling;
+
+import javax.tools.JavaCompiler;
+import javax.tools.ToolProvider;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+import static java.nio.file.StandardWatchEventKinds.ENTRY_CREATE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_DELETE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY;
 
 @SpringBootApplication
+@EnableScheduling
 public class PluginRunnerApplication {
-    
+
+    @Autowired
+    private YAMLConfig myConfig;
+    @Autowired
+    private ApplicationContext ctx;
+    private static ClassLoader PARENT_CLASS_LOADER;
+    private static DynamicClassLoader CLASS_LOADER;
+
     public static void main(String[] args) {
+        PARENT_CLASS_LOADER = DynamicClassLoader.class.getClassLoader();
+        CLASS_LOADER = new DynamicClassLoader(PARENT_CLASS_LOADER);
+        Thread.currentThread().setContextClassLoader(CLASS_LOADER);
         new SpringApplicationBuilder(PluginRunnerApplication.class)
                 .web(WebApplicationType.NONE)
                 .run(args);
     }
-    
+
+    @Scheduled(fixedDelay = Long.MAX_VALUE, initialDelay = 1000)
+    public void reload() throws ClassNotFoundException, IOException, InterruptedException {
+        BeanDefinitionRegistry registry = (BeanDefinitionRegistry) ctx.getAutowireCapableBeanFactory();
+        WatchService watchService = FileSystems.getDefault().newWatchService();
+
+        String pathToProject = System.getProperty("user.dir");
+
+        //get packagename and path to user's filters from YAML file
+        String packageName = myConfig.getPackageName();
+        String absolutePath = myConfig.getPath();
+        if (packageName.equals("")) {
+            packageName = "org.apache.apisix.plugin.runner.filter";
+        }
+        if (absolutePath.equals("")) {
+            absolutePath = pathToProject + "/runner-plugin/src/main/java/org/apache/apisix/plugin/runner/filter/";
+        }
+        Path path = Paths.get(absolutePath);
+
+        //make /target/classes directory if not already exists, compiled java files are output here
+        new File(pathToProject + "/target").mkdirs();
+        new File(pathToProject + "/target/classes").mkdirs();
+
+        //detect changes when files in the path are created, modified, or deleted
+        path.register(watchService, ENTRY_CREATE, ENTRY_MODIFY, ENTRY_DELETE);
+        boolean poll = true;
+        while (poll) {
+            WatchKey key = watchService.take();
+            for (WatchEvent<?> event : key.pollEvents()) {
+                String[] allFilters = new File(absolutePath).list();
```

Comment:
we should check if `allFilters` is empty list

Reply:
done